packages/fast-usdc/src/pool-share-math.js

```diff
@@ -0,0 +1,121 @@
+import { AmountMath } from '@agoric/ertp/src/amountMath.js';
+import {
+  divideBy,
+  makeRatio,
+  makeRatioFromAmounts,
+  multiplyBy,
+} from '@agoric/zoe/src/contractSupport/ratio.js';
+import { Fail, q } from '@endo/errors';
+
+const { getValue, add, isEmpty, isEqual, isGTE, subtract } = AmountMath;
+
+/**
+ * Invariant: shareWorth is the pool balance divided by shares outstanding.
+ *
+ * Use `makeParity(make(USDC, epsilon), PoolShares)` for an initial
+ * value, for some negligible `epsilon` such as 1n.
+ *
+ * @typedef {Ratio} ShareWorth
+ */
+
+/**
+ * Make a 1-to-1 ratio between amounts of 2 brands.
+ *
+ * @param {Amount<'nat'>} numerator
+ * @param {Brand<'nat'>} denominatorBrand
+ */
+export const makeParity = (numerator, denominatorBrand) => {
+  const value = getValue(numerator.brand, numerator);
+  return makeRatio(value, numerator.brand, value, denominatorBrand);
+};
+
+/**
+ * @typedef {{
+ *   deposit: {
+ *     give: { USDC: Amount<'nat'> },
+ *     want?: { PoolShare: Amount<'nat'> }
+ *   },
+ *   withdraw: {
+ *     give: { PoolShare: Amount<'nat'> }
+ *     want: { USDC: Amount<'nat'> },
+ *   }
+ * }} USDCProposalShapes
+ */
+
+/**
+ * Compute Shares payout from a deposit proposal, as well as updated shareWorth.
+ *
+ * Clearly:
+ *
+ * sharesOutstanding' = sharesOutstanding + Shares
+ * poolBalance' = poolBalance + ToPool
+ * shareWorth' = poolBalance' / sharesOutstanding'
+ *
+ * In order to maintain the ShareWorth invariant, we need:
+ *
+ * Shares = ToPool / shareWorth'
+ *
+ * Solving for Shares gives:
+ *
+ * Shares = ToPool * sharesOutstanding / poolBalance
+ *
+ * that is:
+ *
+ * Shares = ToPool / shareWorth
+ *
+ * @param {ShareWorth} shareWorth previous to the deposit
+ * @param {USDCProposalShapes['deposit']} proposal
+ * @returns {{ payouts: { PoolShare: Amount<'nat'> }; shareWorth: ShareWorth }}
+ */
+export const deposit = (shareWorth, { give, want }) => {
+  assert(!isEmpty(give.USDC)); // nice diagnostic provided by proposalShape
+
+  const { denominator: sharesOutstanding, numerator: poolBalance } = shareWorth;
+
+  const PoolShare = divideBy(give.USDC, shareWorth);
+  if (want?.PoolShare) {
+    isGTE(PoolShare, want.PoolShare) ||
+      Fail`deposit cannot pay out ${q(want.PoolShare)}; ${q(give.USDC)} only gets ${q(PoolShare)}`;
+  }
+  const outstandingPost = add(sharesOutstanding, PoolShare);
+  const balancePost = add(poolBalance, give.USDC);
+  const worthPost = makeRatioFromAmounts(balancePost, outstandingPost);
+  return harden({ payouts: { PoolShare }, shareWorth: worthPost });
+};
+
+const isGT = (x, y) => isGTE(x, y) && !isEqual(x, y);
```

Comment:
simpler but you might just inline this:
```suggestion
const isGT = (x, y) => !isGTE(y, x)
```

Reply:
d'oh!

I strongly suspected that `isGT` was over-kill, but I didn't remember why.